src/features/device-state/routes/state-patch-v3.ts

```diff
@@ -0,0 +1,387 @@
+import type { RequestHandler } from 'express';
+
+import * as _ from 'lodash';
+import {
+	captureException,
+	handleHttpErrors,
+} from '../../../infra/error-handling';
+import { sbvrUtils, errors } from '@balena/pinejs';
+import { getIP } from '../../../lib/utils';
+import {
+	Application,
+	Device,
+	Image,
+	ImageInstall,
+	PickDeferred,
+	Release,
+} from '../../../balena-model';
+import type { Filter } from 'pinejs-client-core';
+import { metricsPatchFields, v3ValidPatchFields } from '..';
+import {
+	deleteOldImageInstalls,
+	upsertImageInstall,
+	shouldUpdateMetrics,
+} from '../state-patch-utils';
+
+const { BadRequestError, UnauthorizedError, InternalRequestError } = errors;
+const { api } = sbvrUtils;
+
+/**
+ * These typings should be used as a guide to what should be sent, but cannot be trusted as what actually *is* sent.
+ */
+export type StatePatchV3Body = {
+	[uuid: string]: {
+		name?: string;
+		status?: string;
+		os_version?: string;
+		os_variant?: string;
+		supervisor_version?: string;
+		provisioning_progress?: number | null;
+		provisioning_state?: string | null;
+		ip_address?: string;
+		mac_address?: string;
+		api_port?: number;
+		api_secret?: string;
+		logs_channel?: string; // TODO: should this not be reported anymore?
+		memory_usage?: number;
+		memory_total?: number;
+		storage_block_device?: string;
+		storage_usage?: number;
+		storage_total?: number;
+		cpu_temp?: number;
+		cpu_usage?: number;
+		cpu_id?: string;
+		is_undervolted?: boolean;
+		/**
+		 * Used for setting dependent devices as online
+		 */
+		is_online?: boolean;
+		/**
+		 * Used for setting gateway device of dependent devices
+		 */
+		is_managed_by__device?: number;
```

Comment:
Same thing for this field, the supervisor PR is using `parent_device` for this field (although nothing is really writing this right now).

Reply:
Yeah this is intended to be `is_managed_by__device` to match the model (as we do for other names)